src/include/parser/expression/type_cast_expression.h

```diff
@@ -9,6 +9,9 @@
 namespace noisepage::parser {
 /**
  * TypeCastExpression represents cast expressions of the form CAST(expr) or expr::TYPE.
+ * TypeCastExpression should not exist beyond the optimizer, and the child should be used instead.
+ * The role of a TypeCastExpression is to annotate the type of expr from above in the BinderSherpa.
+ * For example, Postgres does not allow CAST('1+1' AS BIGINT) nor '1+1'::BIGINT, throwing error 22P02.
```

Comment:
This doesn't account for the following trace that works on Postgres

`create table sample (x INTEGER);`
`insert into sample values (1);`
`select x::DECIMAL from sample;`

Reply:
You're right. This actually fails earlier in the binder.